homeassistant/components/sensor/rpi_spi_mcp.py

```diff
@@ -0,0 +1,73 @@
+"""
+Support for analog sensors using RPi SPI via MCP 3008.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.rpi_spi_mcp/
+
+Example configuration:
+"""
+import logging
+import voluptuous as vol
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from spidev import SpiDev
+
+REQUIREMENTS = ['spidev==3.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CHANNELS = 'channels'
+
+CHANNEL_SCHEMA = vol.Schema({
+    vol.Required('name'): cv.string,
+    vol.Required('bus'): cv.positive_int,
+    vol.Required('device'): cv.positive_int,
+    vol.Required('channel'): cv.positive_int,
```

Comment:
How about adding `unit_of_mesurement`?

Reply:
And also the *`device_class`* property, to properly identify the sensor type with metadata...

Reply:
And even a `negate` prop, to return `1 - x` instead of x, which is very useful with some sensors whose voltage drops when the measured magnitude increases.